src/mame/includes/bublbobl.h

```diff
@@ -2,45 +2,60 @@
 // copyright-holders:Chris Moore, Nicola Salmoria
 
 #include "cpu/m6805/m68705.h"
+#include "cpu/m6800/m6801.h"
+#include "cpu/z80/z80.h"
+#include "machine/watchdog.h"
+#include "sound/2203intf.h"
+#include "sound/3526intf.h"
+#include "screen.h"
+#include "speaker.h"
 
-#include "machine/gen_latch.h"
 #include "machine/taito68705interface.h"
 
 class bublbobl_state : public driver_device
 {
 public:
 	enum
 	{
-		TIMER_NMI,
 		TIMER_M68705_IRQ_ACK
 	};
 
 	bublbobl_state(const machine_config &mconfig, device_type type, const char *tag)
-		: driver_device(mconfig, type, tag),
-		m_videoram(*this, "videoram"),
-		m_objectram(*this, "objectram"),
-		m_mcu_sharedram(*this, "mcu_sharedram"),
-		m_maincpu(*this, "maincpu"),
-		m_mcu(*this, "mcu"),
-		m_audiocpu(*this, "audiocpu"),
-		m_slave(*this, "slave"),
-		m_gfxdecode(*this, "gfxdecode"),
-		m_palette(*this, "palette"),
-		m_soundlatch(*this, "soundlatch")
+		: driver_device(mconfig, type, tag)
+		, m_videoram(*this, "videoram")
+		, m_objectram(*this, "objectram")
+		, m_mcu_sharedram(*this, "mcu_sharedram")
+		, m_maincpu(*this, "maincpu")
+		, m_mcu(*this, "mcu")
+		, m_audiocpu(*this, "audiocpu")
+		, m_subcpu(*this, "subcpu")
+		, m_screen(*this, "screen")
+		, m_gfxdecode(*this, "gfxdecode")
+		, m_palette(*this, "palette")
+		, m_ym2203(*this, "ym2203")
+		, m_ym3526(*this, "ym3526")
 	{ }
 
 	/* memory pointers */
 	required_shared_ptr<uint8_t> m_videoram;
 	required_shared_ptr<uint8_t> m_objectram;
 	optional_shared_ptr<uint8_t> m_mcu_sharedram;
 
+	/* general */
+	bool     m_is_tokio_hw;
+
 	/* video-related */
-	int      m_video_enable;
+	bool     m_video_enable;
 
 	/* sound-related */
-	int      m_sound_nmi_enable;
-	int      m_pending_nmi;
-	int      m_sound_status;
+	bool     m_sound_nmi_enable;
+	uint8_t  m_fromMain;
+	uint8_t  m_fromSound;
+	bool     m_MainHasWritten;
```

Comment:
m_sound_nmi_enable is a pre-existing variable name from before I started messing around, but I'll make it consistent with the others.